apps/proposer/src/proposer.mjs

```diff
@@ -56,6 +57,32 @@ async function checkAndChangeProposer(nf3) {
   }
 }
 
+/**
+ * check that proposer is registered and register if its not
+ */
+async function checkAndRegisterProposer(nf3, proposerBaseUrl) {
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const { proposers } = await nf3.getProposers();
+    const thisProposer = proposers.filter(p => p.thisAddress === nf3.ethereumAddress);
+    if (!thisProposer.length) {
+      logger.info('Attempting to register proposer');
+      const blockStake = await nf3.getBlockStake();
+      const minimumStake = await nf3.getMinimumStake();
+
+      logger.info(`blockStake: ${blockStake}, minimumStake: ${minimumStake}`);
+
+      try {
+        await nf3.registerProposer(proposerBaseUrl, minimumStake);
+      } catch (err) {
+        logger.info(err);
+      }
+    }
+
+    await new Promise(resolve => setTimeout(resolve, CHECK_REGISTER_PROPOSER_SECOND * 1000));
```

Comment:
The function `waitForTimeout` from `common-files/utils/utils.mjs` can be used here.